cmd/dockerd/daemon_solaris.go

```diff
@@ -0,0 +1,69 @@
+// +build solaris
+
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"syscall"
+
+	"github.com/docker/docker/libcontainerd"
+	"github.com/docker/docker/pkg/system"
+)
+
+const defaultDaemonConfigFile = ""
+
+// currentUserIsOwner checks whether the current user is the owner of the given
+// file.
+func currentUserIsOwner(f string) bool {
```

Comment:
(*thinking out load*) `currentUserIsOwner`, `setDefaultUnmask` and `getLibcontainerdRoot` are exactly the same as what's in `daemon_unix.go`. This is probably a nit, but as Solaris is a UNIX system, I would definitely not add `!solaris` in `daemon_unix.go` and move the non common unix into other files.. but on the other hand it might get worse that way :sweat: